src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -0,0 +1,73 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.tool.ResponseSerializer;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import org.mapdb.*;
+import com.google.gson.Gson;
+import java.util.Map;
+import java.util.HashMap;
+
+@WebServlet("/recommendation")
+public class RecommendationServlet extends HttpServlet {
+  private DB db;
+  private BTreeMap<String, String[]> queryNearestNeighbors;
+  private Map<String, String> jsonErrorInfo;
```

Comment:
Why are we storing this in the class?

Reply:
In order for the variables to be accessed by the init() and destroy() function. I only want to open the db and grab the queryNearestNeighbors one time. Though looking back, jsonErrorInfo can be put in inside the method call